server/api.go

```diff
@@ -931,3 +942,124 @@ func (p *Plugin) updateSettings(w http.ResponseWriter, r *http.Request) {
 	resp, _ := json.Marshal(info.Settings)
 	w.Write(resp)
 }
+
+func (p *Plugin) getRepositories(w http.ResponseWriter, r *http.Request) {
+	userID := r.Header.Get("Mattermost-User-ID")
+	if userID == "" {
+		http.Error(w, "Not authorized", http.StatusUnauthorized)
+		return
+	}
+
+	info, err := p.getGitHubUserInfo(userID)
+	if err != nil {
+		writeAPIError(w, err)
+		return
+	}
+
+	ctx := context.Background()
+	githubClient := p.githubConnect(*info.Token)
+
+	repositories, _, apiErr := githubClient.Repositories.List(ctx, "", &github.RepositoryListOptions{})
+	if apiErr != nil {
+		mlog.Error(apiErr.Error())
+		return
+	}
+
+	resp, _ := json.Marshal(repositories)
+	w.Write(resp)
+}
+
+func (p *Plugin) createIssue(w http.ResponseWriter, r *http.Request) {
+	if r.Method != http.MethodPost {
+		http.Error(w, fmt.Sprintf("Request: %s is not allowed, must be POST", r.Method), http.StatusMethodNotAllowed)
+		return
+	}
+
+	userID := r.Header.Get("Mattermost-User-ID")
+	if userID == "" {
+		http.Error(w, "Not authorized", http.StatusUnauthorized)
+		return
+	}
+
+	// get data for the issue from the request body and fill IssueRequest object
+	issue := &IssueRequest{}
+	dec := json.NewDecoder(r.Body)
+	if err := dec.Decode(&issue); err != nil {
+		mlog.Error("Error decoding JSON body", mlog.Err(err))
+		writeAPIError(w, &APIErrorResponse{ID: "", Message: "Please provide a JSON object.", StatusCode: http.StatusBadRequest})
+		return
+	}
+
+	if issue.Title == "" {
+		writeAPIError(w, &APIErrorResponse{ID: "", Message: "Please provide a valid issue title.", StatusCode: http.StatusBadRequest})
+		return
+	}
+
+	if issue.Body == "" {
+		writeAPIError(w, &APIErrorResponse{ID: "", Message: "Please provide a valid issue description.", StatusCode: http.StatusBadRequest})
+		return
+	}
+
+	if issue.Repo == "" {
+		writeAPIError(w, &APIErrorResponse{ID: "", Message: "Please provide a valid repo name.", StatusCode: http.StatusBadRequest})
+		return
+	}
+	ctx := context.Background()
+	gh_issue := &github.IssueRequest{
+		Title: &issue.Title,
+		Body:  &issue.Body,
+	}
+
+	// Make sure user have a connected github account
+	info, err := p.getGitHubUserInfo(userID)
+	if err != nil {
+		writeAPIError(w, err)
+		return
+	}
+
+	githubClient := p.githubConnect(*info.Token)
+
+	// call Create(ctx context.Context, owner string, repo string, issue *IssueRequest)
+	result, resp, api_err := githubClient.Issues.Create(ctx, info.GitHubUsername, issue.Repo, gh_issue)
+	if api_err != nil {
+		mlog.Error(api_err.Error())
```

Comment:
Why does the code continue if there is an error?

Reply:
Any apiErr from the client now handled with writeAPIError. https://github.com/mattermost/mattermost-plugin-github/pull/239/commits/d2a20760d413d4b4020afeff3edcce728a2dedf6